src/data/queries/banners.ts

```diff
@@ -0,0 +1,57 @@
+import { QueryFormatter } from 'next-drupal-query'
+import { NodeBanner } from '@/types/dataTypes/drupal/node'
+import { BannerType, FacilityBannerType, PromoBannerType } from '@/types/index'
+
+export const BannerDisplayType = {
+  PROMO_BANNER: 'promoBanner',
+  FACILITY_BANNER: 'facilityBanner',
+  BANNER: 'banner',
+}
+
+export const BannerTypeMapping = {
+  [BannerDisplayType.PROMO_BANNER]: 'node--promo_banner',
+  [BannerDisplayType.FACILITY_BANNER]: 'node--full_width_banner_alert',
+  [BannerDisplayType.BANNER]: 'node--banner',
+}
+
+export const formatter: QueryFormatter<
+  NodeBanner,
+  PromoBannerType | BannerType | FacilityBannerType
+> = (entity: NodeBanner) => {
+  return entity.bannerData.map((banner) => {
```

Comment:
Let's add tests for input/output checking; there are examples for the expandableText query

Reply:
done!